### PR TITLE
Sorted list

### PR DESCRIPTION
**Pull request by bertoia**
Sorts the view by deadline (Tasks), start time (Events) and last modified (Floating Tasks).
Added a lastModifiedTime field for Entry which is updated by the edit/tag/untag/mark/unmark commands.

There are quite a few changes to the tests, especially those that checks for task ordering.
Might want to check them out to be familiar with how sorted entries affect the test cases.

clear PriorityQ.xml before running the app as lastModifiedTime is a compulsory field.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 1]
```diff
@@ -77,6 +78,7 @@ public CommandResult execute() {
             }
 
             taskToEdit = lastShownList.get(targetIndex - 1);
+            originalLastModifiedTime = taskToEdit.getLastModifiedTime().plusDays(0);
```
Comment (tjonganthony) [inline src/main/java/seedu/address/logic/commands/EditCommand.java]: LocalDateTime is an immutable field, so don't worry about copying reference.

File: src/main/java/seedu/address/logic/commands/MarkCommand.java [hunk 2]
```diff
@@ -39,6 +42,7 @@ public CommandResult execute() {
             }
     
             entryToMark = lastShownList.get(targetIndex - 1);
+            originalLastModifiedTime = entryToMark.getLastModifiedTime().plusDays(0);
```
Comment (tjonganthony) [inline src/main/java/seedu/address/logic/commands/MarkCommand.java]: same as before.

File: src/main/java/seedu/address/logic/commands/TagCommand.java [hunk 3]
```diff
@@ -57,6 +59,7 @@ public CommandResult execute() {
             }
             
             taskToTag = lastShownList.get(targetIndex - 1);
+            originalLastModifiedTime = taskToTag.getLastModifiedTime().plusDays(0);
```
Comment (tjonganthony) [inline src/main/java/seedu/address/logic/commands/TagCommand.java]: same again.

File: src/main/java/seedu/address/logic/commands/UnmarkCommand.java [hunk 4]
```diff
@@ -39,6 +42,7 @@ public CommandResult execute() {
             }
     
             entryToUnmark = lastShownList.get(targetIndex - 1);
+            originalLastModifiedTime = entryToUnmark.getLastModifiedTime().plusDays(0);
```
Comment (tjonganthony) [inline src/main/java/seedu/address/logic/commands/UnmarkCommand.java]: same here

File: src/main/java/seedu/address/logic/commands/UntagCommand.java [hunk 5]
```diff
@@ -57,6 +59,7 @@ public CommandResult execute() {
             }
             
             taskToUntag = lastShownList.get(targetIndex - 1);
+            originalLastModifiedTime = taskToUntag.getLastModifiedTime().plusDays(0);
```
Comment (tjonganthony) [inline src/main/java/seedu/address/logic/commands/UntagCommand.java]: same.

File: src/main/java/seedu/address/model/task/Entry.java [hunk 6]
```diff
@@ -184,4 +188,39 @@ public final String tagsString() {
             return buffer.substring(0, buffer.length() - DELIMITER.length());
         }
     }
+    
+    //@@author A0121501E
+    public int compareTo(Entry o) {
```
Comment (tjonganthony) [inline src/main/java/seedu/address/model/task/Entry.java]: Don't implement the compareTo function. It should be implemented by the class that extends Entry.

File: src/main/java/seedu/address/model/task/Task.java [hunk 7]
```diff
@@ -36,21 +36,22 @@ public Task(Title title, LocalDateTime deadline, UniqueTagList tags, boolean isM
...
         }
+        this.lastModifiedTime = new SimpleObjectProperty<>(lastModifiedTime);
     }
 
     /**
      * Every field must be present and not null.
      */
     public Task(Title title, UniqueTagList tags) {
-        this(title, null, tags, false, "");
+        this(title, null, tags, false, "", LocalDateTime.MIN);
```
Comment (tjonganthony) [inline src/main/java/seedu/address/model/task/Task.java:46]: Why is this MIN, not now?

File: src/main/java/seedu/address/model/task/Entry.java (hunk 6 above)
Comment (bertoia) [reply src/main/java/seedu/address/model/task/Entry.java]: Why is it better to implement it in the child class?

File: src/main/java/seedu/address/logic/commands/EditCommand.java (hunk 1 above)
Comment (bertoia) [reply src/main/java/seedu/address/logic/commands/EditCommand.java]: Changed.

File: src/main/java/seedu/address/model/task/Task.java (hunk 7 above)
Comment (bertoia) [reply src/main/java/seedu/address/model/task/Task.java:46]: This constructor is only used by tests. I am not sure if the tests will consistently pass/fail if this field is allowed to vary across runs.

Comment (tjonganthony) [reply src/main/java/seedu/address/model/task/Task.java:46]: Oh , good observation!

File: src/main/java/seedu/address/model/task/Entry.java (hunk 6 above)
Comment (tjonganthony) [reply src/main/java/seedu/address/model/task/Entry.java]: Hmmm, it's a bit hard to explain.
In short, the compareTo function is not meant for this.
Remove the `compareTo` function and `implement Comparable<Entry>`.
Add a function called `compareShowDate` or something instead.

Comment (bertoia) [reply src/main/java/seedu/address/model/task/Entry.java]: Ahh I see. I will use a comparator instead.

**Comment by tjonganthony** [comment]
lgtm.

**Pull request merged**